implants/lib/eldritch/src/sys/write_reg_impl.rs

```diff
@@ -0,0 +1,209 @@
+use anyhow::Result;
+use hex;
+
+
+pub fn write_reg(reghive: String, regpath: String, regname: String, regtype:String, regvalue: String ) -> Result<String>  {
+    
+    #[cfg(not(target_os = "windows"))]
+        return Err(anyhow::anyhow!("This OS isn't supported by the write_reg function. Only windows systems are supported"));
+
+    #[cfg(target_os = "windows")]{
+        use winreg::{{enums::*}, RegKey, RegValue};
+
+        //Accepted values for reghive :
+        //HKEY_CLASSES_ROOT, HKEY_CURRENT_USER, HKEY_LOCAL_MACHINE, HKEY_USERS, HKEY_PERFORMANCE_DATA, HKEY_PERFORMANCE_TEXT, HKEY_PERFORMANCE_NLSTEXT, HKEY_CURRENT_CONFIG, HKEY_DYN_DATA, HKEY_CURRENT_USER_LOCAL_SETTINGS
+        let ihive: isize = match reghive.as_ref() {
+            "HKEY_CLASSES_ROOT" => HKEY_CLASSES_ROOT,
+            "HKEY_CURRENT_USER" => HKEY_CURRENT_USER,
+            "HKEY_LOCAL_MACHINE" => HKEY_LOCAL_MACHINE,
+            "HKEY_USERS" => HKEY_USERS,
+            "HKEY_PERFORMANCE_DATA" => HKEY_PERFORMANCE_DATA,
+            "HKEY_PERFORMANCE_TEXT" => HKEY_PERFORMANCE_TEXT,
+            "HKEY_PERFORMANCE_NLSTEXT" => HKEY_PERFORMANCE_NLSTEXT,
+            "HKEY_CURRENT_CONFIG" => HKEY_CURRENT_CONFIG,
+            "HKEY_DYN_DATA" => HKEY_DYN_DATA,
+            "HKEY_CURRENT_USER_LOCAL_SETTINGS" => HKEY_CURRENT_USER_LOCAL_SETTINGS,
+            _ => return Err(anyhow::anyhow!("RegHive can only be one of the following values - HKEY_CLASSES_ROOT, HKEY_CURRENT_USER, HKEY_LOCAL_MACHINE, HKEY_USERS, HKEY_PERFORMANCE_DATA, HKEY_PERFORMANCE_TEXT, HKEY_PERFORMANCE_NLSTEXT, HKEY_CURRENT_CONFIG, HKEY_DYN_DATA, HKEY_CURRENT_USER_LOCAL_SETTINGS ")),
+            
+        };
+        
+        let hive = RegKey::predef(ihive);
+        let (nkey, ndisp) = hive.create_subkey(regpath)?;
+
+        let resp : String = match ndisp {
+            REG_CREATED_NEW_KEY => "A new key has been created".to_string(),
+            REG_OPENED_EXISTING_KEY => "An existing key has been modified".to_string(),
+        };
+
+        match regtype.as_ref() {
+            "REG_NONE" => {
+                // Assume value is a plain string
+                nkey.set_value(regname, &regvalue)?;
+            },
+            "REG_SZ" => nkey.set_value(regname, &regvalue)?,
+            "REG_EXPAND_SZ" => nkey.set_value(regname, &regvalue)?,
+            "REG_BINARY" => {
+                // Assume value is a hexadecimal string (e.g., "deadbeef")
+                let parsed_value: Vec<u8> = hex::decode(regvalue).expect("Invalid REG_BINARY value");
```

Comment:
Expect will panic and we'd like to bubble the error up so golem or imix can handle it.
If hex::decode returns a Result you can use the `?` operator.
If it returns an option try the `anyhow::Context` function. 
```rust
hex::decode(regvalue).context("Invalid REG_BINARY value")?;
```

Context will take an option and turn it into a Result or Error using the provided string.
Then we can bubble it up with `?`